src/registers/mxcsr.rs

```diff
@@ -0,0 +1,107 @@
+//! Functions to read and write MXCSR register.
+
+#[cfg(feature = "instructions")]
+pub use self::x86_64::*;
+
+use bitflags::bitflags;
+
+bitflags! {
+    /// MXCSR register.
+    #[repr(transparent)]
+    pub struct MxCsr: u32 {
+        /// Invalid operation flag
+        const INVALID_OPERATION = 1 << 0;
+        /// Denormal flag
+        const DENORMAL = 1 << 1;
+        /// Divide-by-zero flag
+        const DIVIDE_BY_ZERO = 1 << 2;
+        /// Overflow flag
+        const OVERFLOW = 1 << 3;
+        /// Underflow flag
+        const UNDERFLOW = 1 << 4;
+        /// Precision flag
+        const PRECISION = 1 << 5;
+        /// Denormals are zeros
+        const DENORMALS_ARE_ZEROS = 1 << 6;
+        /// Invalid operation mask
+        const INVALID_OPERATION_MASK = 1 << 7;
+        /// Denormal mask
+        const DENORMAL_MASK = 1 << 8;
+        /// Divide-by-zero mask
+        const DIVIDE_BY_ZERO_MASK = 1 << 9;
+        /// Overflow mask
+        const OVERFLOW_MASK = 1 << 10;
+        /// Underflow mask
+        const UNDERFLOW_MASK = 1 << 11;
+        /// Precision mask
+        const PRECISION_MASK = 1 << 12;
+        /// Toward negative infinity
+        const ROUNDING_CONTROL_NEGATIVE = 1 << 13;
+        /// Toward positive infinity
+        ///
+        /// Set the both control registers for "toward zero".
+        const ROUNDING_CONTROL_POSITIVE = 1 << 14;
+        /// Flush to zero
+        const FLUSH_TO_ZERO = 1 << 15;
+    }
+}
+
+impl Default for MxCsr {
+    /// Return the default MXCSR value at reset, as documented in Intel SDM volume 2A.
+    #[inline]
+    fn default() -> Self {
+        MxCsr::from_bits_truncate(0x1F80)
```

Comment:
This values should probably be generated by combining the individual flags with the `|` operator. It might still be useful to mention that constant in a comment somewhere though.

Reply:
I can do it that way. Yeah, it is still useful to have also `0x1F80` somewhere. E.g. I sometimes use some constant values that I remember as search keys for the SDM :-D I'll change this as the disjunction of the flags and mention the constant in a comment.